Assets/BossRoom/Scripts/Client/UI/Lobby/LobbyUIMediator.cs

```diff
@@ -0,0 +1,338 @@
+        {
+            m_LobbyAsyncRequests.CreateLobbyAsync(lobbyName, maxPlayers, isPrivate, onlineMode, ip, port, OnCreatedLobby, OnFailedLobbyCreateOrJoin);
+            BlockUIWhileLoadingIsInProgress();
+        }
+
+        public void QueryLobbiesRequest(bool blockUI)
+        {
+
+            m_LobbyAsyncRequests.RetrieveLobbyListAsync(
+                OnSuccess,
+                OnFailure
+            );
+
+            if (blockUI)
+            {
+                BlockUIWhileLoadingIsInProgress();
+            }
+
+            void OnSuccess(QueryResponse qr)
+            {
+                UnblockUIAfterLoadingIsComplete();
+                var localLobbies = LocalLobby.CreateLocalLobbies(qr);
+
+                var newLobbyDict = new Dictionary<string, LocalLobby>();
+
+                foreach (var lobby in localLobbies)
+                {
+                    newLobbyDict.Add(lobby.LobbyID, lobby);
+                }
+
+                m_lobbyServiceData.FetchedLobbies(newLobbyDict);
+            }
+
+            void OnFailure()
+            {
+                UnblockUIAfterLoadingIsComplete();
+            }
+        }
+
+        public void JoinLobbyWithCodeRequest(string lobbyCode)
+        {
+            m_LobbyAsyncRequests.JoinLobbyAsync(null, lobbyCode, OnJoinedLobby, OnFailedLobbyCreateOrJoin);
+            BlockUIWhileLoadingIsInProgress();
+        }
+
+        public void JoinLobbyRequest(LocalLobby lobby)
+        {
+            m_LobbyAsyncRequests.JoinLobbyAsync(lobby.LobbyID, lobby.LobbyCode, OnJoinedLobby, OnFailedLobbyCreateOrJoin);
+            BlockUIWhileLoadingIsInProgress();
+        }
+
+        public void QuickJoinRequest()
+        {
+            m_LobbyAsyncRequests.QuickJoinLobbyAsync(m_localUser, OnJoinedLobby, OnFailedLobbyCreateOrJoin);
+            BlockUIWhileLoadingIsInProgress();
+        }
+
+        private void OnFailedLobbyCreateOrJoin()
+        {
+            UnblockUIAfterLoadingIsComplete();
+        }
+
+        private void OnCreatedLobby(Lobby r)
+        {
+            m_localLobby.ApplyRemoteData(r);
+            m_localUser.IsHost = true;
+
+            m_LobbyAsyncRequests.BeginTracking(r);
+            m_lobbyContentHeartbeat.BeginTracking();
+
+            switch (m_localLobby.OnlineMode)
+            {
+                case OnlineMode.IpHost:
+                    Debug.Log($"Created lobby with ID: {m_localLobby.LobbyID} and code {m_localLobby.LobbyCode}, at IP:Port {m_localLobby.Data.IP}:{m_localLobby.Data.Port}");
+                    break;
+                case OnlineMode.UnityRelay:
+                    Debug.Log($"Created lobby with ID: {m_localLobby.LobbyID} and code {m_localLobby.LobbyCode}, Internal Relay Join Code{m_localLobby.RelayJoinCode}");
+                    break;
+            }
+
+            m_GameNetPortal.PlayerName = m_localUser.DisplayName;
+
+            var cancellationTokenSource = new CancellationTokenSource();
```

Comment:
Since this cancellation token source is only scoped to this method, it looks like it can't be cancelled. Since cancelling seems to no longer be an option, this token should be simply removed.
However, if cancelling is still something we want, this token source should have a broader scope to allow calling cancellationTokenSource.Cancel() when cancelling.

Reply:
Same thing for the token source in the OnJoinedLobby method

Reply:
created https://jira.unity3d.com/browse/MTT-2697